src/util/SystemInfo/SystemInfoUnix.cpp

```diff
@@ -0,0 +1,56 @@
+#if BOOST_OS_LINUX
+
+#include "util/SystemInfo/SystemInfo.h"
+
+#include <unistd.h>
+#include <sys/times.h>
+
+#include <string>
+
+uint32 GetProcessorCount()
+{
+	return std::thread::hardware_concurrency();
+}
+
+uint64 QueryRamUsage()
+{
+	long page_size = sysconf(_SC_PAGESIZE);
+
+	std::ifstream file("/proc/self/statm");
```

Comment:
You should look into using `getrusage`, it's supported on Linux, OSX, FreeBSD and OpenBSD
https://man7.org/linux/man-pages/man2/getrusage.2.html
https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/getrusage.2.html
https://www.freebsd.org/cgi/man.cgi?query=getrusage&sektion=2
https://man.openbsd.org/getrusage.2

Reply:
I know about `getrusage` but I don't think you can get the current memory usage from that. Please let me know if it is possible.

Reply:
I think you just need to read `ru_maxrss`

```
ru_maxrss (since Linux 2.6.32)
              This is the maximum resident set size used (in kilobytes).  For RUSAGE_CHILDREN, this is the resident set size of the largest child, not the maximum resident set size of the process tree.
```

Reply:
Oh I see the issue, it's the maximum value on the lifetime of the process isn't it?